src/wmtk/TriMesh.cpp

```diff
@@ -337,11 +337,12 @@ bool TriMesh::is_connectivity_valid() const
         }
         int cnt = 0;
         for (long j = 0; j < 3; ++j) {
-            if (fe_accessor.index_access().vector_attribute(ef_accessor.index_access().scalar_attribute(i))[j] == i) {
+            if (fe_accessor.index_access().vector_attribute(
+                    ef_accessor.index_access().scalar_attribute(i))[j] == i) {
                 cnt++;
             }
         }
-        if (cnt != 1) {
+        if (cnt == 0) {
```

Comment:
I assume these changes are not related to this PR. Did you maybe pull another branch into this one? It might make sense to pull the main branch into this one once more to clean up changes that are actually from some other branch.

Reply:
I created an issue for this change. It is actually not determined yet (check for glued edges in trimesh). 
I redo the code in this PR

Reply:
I cannot redo this change. Many tests in 2d need this change. It is not from another branch. We did this change when we realized the corner case in 2d is not handled as it was in 1d so we did this change here.